components/server/src/oauth-server/db.ts

```diff
@@ -141,6 +141,17 @@ const desktopClient: OAuthClient = {
     ],
 };
 
+const toolbox: OAuthClient = {
+    id: "toolbox-gateway-gitpod-plugin",
+    name: "JetBrains Toolbox Gitpod Plugin",
+    redirectUris: ["jetbrains://gateway/io.gitpod.toolbox.gateway/auth"],
+    allowedGrants: ["authorization_code"],
+    scopes: [
+        // We scope all so that it can work in papi like a PAT
+        { name: "function:*" },
```

Comment:
💭 Is it necessary to whitelist all scopes or can we limit it to some extent of internal functions, like we did for the other clients?

Reply:
Yes we do, because we need to use papi v1 entry. https://github.com/gitpod-io/gitpod/pull/19597#discussion_r1557720549

We can update it anyway in the future

Reply:
For more context, the limitation is here https://github.com/gitpod-io/gitpod/blob/8df5d2f04880e8d05c3c2ec9a42219907e7741b3/components/server/src/auth/bearer-authenticator.ts#L115-L119

Reply:
But I have no idea when we can do it